spec/jobs/reports/combined_invoice_supplement_report_v2_spec.rb

```diff
@@ -8,341 +8,561 @@
     ServiceProvider.delete_all
   end
 
-  let(:partner_account1) { create(:partner_account) }
-  let(:partner_account2) { create(:partner_account) }
-  let(:gtc1) do
-    create(
-      :iaa_gtc,
-      gtc_number: 'gtc1234',
-      partner_account: partner_account1,
-      start_date: iaa1_range.begin,
-      end_date: iaa1_range.end,
-    )
-  end
+  describe '#perform' do
+    it 'is empty with no data' do
+      csv = CSV.parse(report.perform(Time.zone.today), headers: true)
+      expect(csv).to be_empty
+    end
 
-  let(:gtc2) do
-    create(
-      :iaa_gtc,
-      gtc_number: 'gtc5678',
-      partner_account: partner_account2,
-      start_date: iaa2_range.begin,
-      end_date: iaa2_range.end,
-    )
-  end
+    context 'with data generates reports by iaa + order number, issuer and year_month' do
+      let(:user1) { create(:user, profiles: [profile1]) }
+      let(:profile1) { build(:profile, verified_at: DateTime.new(2019, 10, 15).utc) }
 
-  let(:iaa_order1) do
-    build_iaa_order(order_number: 1, date_range: iaa1_range, iaa_gtc: gtc1)
-  end
-  let(:iaa_order2) do
-    build_iaa_order(order_number: 2, date_range: iaa2_range, iaa_gtc: gtc2)
-  end
+      context 'iaa1 single issuer in year_month April 2020' do
```

Comment:
so I think it's good that we've split up the test so that each context has its own IAA.

Now that that is happening, "iaa1" "iaa2" "iaa3" don't really mean anything? They won't exist at the same time, so we can just describe what's going on:
```suggestion
      context 'with an IAA with a single issuer in April 2020' do
```